README.md
# flat_banners

[![pub package](https://img.shields.io/pub/v/flat_banners.svg)](https://pub.dev/packages/flat_banners)

A Flutter package for Offers, Promo and Invites banners. Android support
Android and Web.

## Usage

To use this plugin, add `flat_banners` as a [dependency in your pubspec.yaml file](https://flutter.dev/platform-plugins/).

## Demo

<img src="https://github.com/om-chauhan/flat_banners/raw/master/banners-package-demo.gif" height="500" />

### Example

```dart
import 'package:flutter/material.dart';
import 'package:flat_banners/flat_banners.dart';
import 'package:example/controller/data_controller.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Banners Example',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return Center(
      child: FlatBanners(
        imageWidth: 50,
        gradientColors: [
          const Color(0xff6C59D4).withOpacity(0.9),
          const Color(0xff869DFB).withOpacity(0.7),
        ],
        title: 'Get 500 Coins by Referring',
        subtitle: 'Games are Fun with Friends!',
        btnText: 'Refer a friend',
        image: 'assets/icons/social-media.png',
        onPressed: () {},
      ),
    );
  }
}
```

See the example app for more complex examples.

## Screenshot

|                                       Horizontal View|                                       Vertical View                               |
| :-----------------------------------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------------------: |
| <img src="https://github.com/om-chauhan/flat_banners/raw/master/horizontal-vertical-view.jpg" height="400" /> | <img src="https://github.com/om-chauhan/flat_banners/raw/master/vertical-view.jpg" height="400" /> |


CHANGELOG.md
## 0.0.1

- initial release.

## 0.0.2

- demo gif and screenshot added in `Readme.md`

## 0.0.3

- update `Readme.md`

## 0.0.5

- update `Readme.md`

## 0.0.6

- update `Readme.md`


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flat_banners/flat_banners.dart';
import 'package:example/controller/data_controller.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Banners Example',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    final data = DataController();
    return Scaffold(
      appBar: AppBar(
        title: Text('Banners Example'),
      ),
      body: SingleChildScrollView(
        padding: EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 10.0),
            Text(
              'ListView Builder (Horizontal Scrolling)',
              style: TextStyle(
                fontSize: 18,
              ),
            ),
            const SizedBox(height: 10.0),
            Container(
              height: 105,
              child: ListView.builder(
                itemCount: data.dataList.length,
                shrinkWrap: true,
                scrollDirection: Axis.horizontal,
                physics: ScrollPhysics(),
                itemBuilder: (BuildContext context, int index) {
                  return Padding(
                    padding: const EdgeInsets.only(right: 8.0),
                    child: FlatBanners(
                      onPressed: () {},
                      subtitle: data.dataList[index].subTitle,
                      title: data.dataList[index].title,
                      btnText: data.dataList[index].actionText,
                      image: data.dataList[index].image,
                      imageWidth: 50,
                      gradientColors: data.dataList[index].gradientColors,
                    ),
                  );
                },
              ),
            ),
            const SizedBox(height: 10.0),
            Text(
              'ListView Builder (Vertical Scrolling)',
              style: TextStyle(
                fontSize: 18,
              ),
            ),
            const SizedBox(height: 10.0),
            ListView.builder(
              itemCount: data.dataList.length,
              shrinkWrap: true,
              physics: ScrollPhysics(),
              itemBuilder: (BuildContext context, int index) {
                return Padding(
                  padding: const EdgeInsets.only(top: 8, bottom: 8),
                  child: FlatBanners(
                    onPressed: () {},
                    subtitle: data.dataList[index].subTitle,
                    title: data.dataList[index].title,
                    btnText: data.dataList[index].actionText,
                    image: data.dataList[index].image,
                    imageWidth: 50,
                    gradientColors: data.dataList[index].gradientColors,
                  ),
                );
              },
            )
          ],
        ),
      ),
    );
  }
}


lib/src/flat_banners.dart
import 'package:flutter/material.dart';

class FlatBanners extends StatelessWidget {
  const FlatBanners({
    Key? key,
    this.subtitle,
    this.title,
    this.image,
    this.btnText,
    this.gradientColors = const [
      Colors.blueGrey,
      Colors.blueAccent,
    ],
    this.imageHeight,
    this.imageWidth = 65,
    this.startGradientAlignment = Alignment.topLeft,
    this.endGradientAlignment = Alignment.bottomRight,
    this.actionBtnBgColor = const Color(0xff5957AB),
    this.onPressed,
  }) : super(key: key);
  final String? subtitle;
  final String? title;
  final String? image;
  final String? btnText;
  final Color? actionBtnBgColor;
  final List<Color> gradientColors;
  final double? imageHeight;
  final double? imageWidth;
  final AlignmentGeometry startGradientAlignment;
  final AlignmentGeometry endGradientAlignment;
  final Function()? onPressed;
  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onPressed,
      child: Container(
        decoration: BoxDecoration(
          color: Colors.green,
          borderRadius: BorderRadius.circular(12),
          gradient: LinearGradient(
            colors: gradientColors,
            begin: startGradientAlignment,
            end: endGradientAlignment,
          ),
        ),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Padding(
              padding: const EdgeInsets.all(10),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    subtitle ?? 'Subtitle goes here...',
                    style: TextStyle(
                      color: Colors.white.withOpacity(0.7),
                      fontSize: 12,
                    ),
                  ),
                  Text(
                    title ?? 'Title goes here....',
                    style: const TextStyle(
                      color: Colors.white,
                      fontSize: 18,
                      fontWeight: FontWeight.w500,
                    ),
                  ),
                  const SizedBox(height: 10.0),
                  Container(
                    padding: const EdgeInsets.symmetric(
                        horizontal: 30, vertical: 10),
                    decoration: BoxDecoration(
                      color: actionBtnBgColor,
                      borderRadius: BorderRadius.circular(30),
                    ),
                    child: Text(
                      btnText ?? 'Action text...',
                      style: const TextStyle(
                        color: Colors.white,
                        letterSpacing: 1,
                        fontWeight: FontWeight.w500,
                        fontSize: 14,
                      ),
                    ),
                  ),
                ],
              ),
            ),
            const SizedBox(width: 10.0),
            Padding(
              padding: const EdgeInsets.only(right: 10.0),
              child: Image.asset(
                image ?? 'assets/icons/photo.png',
                height: imageHeight,
                width: imageWidth,
                fit: BoxFit.cover,
              ),
            ),
          ],
        ),
      ),
    );
  }
}


example/lib/controller/data_controller.dart
import 'package:example/model/data_model.dart';
import 'package:flutter/material.dart';

class DataController {
  final dataList = [
    DataModel(
      title: 'Get 500 Coins by Referring',
      subTitle: 'Games are Fun with Friends!',
      image: 'assets/icons/social-media.png',
      actionText: 'Refer a friend',
      actionBgColor: const Color(0xff5957AB),
      gradientColors: [
        const Color(0xff6C59D4).withOpacity(0.9),
        const Color(0xff869DFB).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Get 1,000 Coins to play!',
      subTitle: 'Want to get better and win more!',
      image: 'assets/icons/coins.png',
      actionText: 'Try Practice Leagues',
      actionBgColor: const Color(0xff5D808C),
      gradientColors: [
        const Color(0xff48A2EC).withOpacity(0.9),
        const Color(0xffEDD177).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Get Your Diwali Gift',
      subTitle: 'Celebrate Diwali with Friends!',
      image: 'assets/icons/giftbox.png',
      actionText: 'Claim Now',
      actionBgColor: const Color(0xff4C3063),
      gradientColors: [
        const Color(0xff270047).withOpacity(0.9),
        const Color(0xff3B0148).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Results Announced',
      subTitle: 'Shubh Festival Contest - Edition 1',
      image: 'assets/icons/trophy.png',
      actionText: 'View Winners',
      actionBgColor: const Color(0xff844595),
      gradientColors: [
        const Color(0xff8D2195).withOpacity(0.9),
        const Color(0xff3D297B).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Candy Bust is now available',
      subTitle: 'Are you a candy crush player ?',
      image: 'assets/icons/stars.png',
      actionText: 'Play Now',
      actionBgColor: const Color(0xffBE6692),
      gradientColors: [
        const Color(0xffFB83C6).withOpacity(0.9),
        const Color(0xffFFD65E).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Join now with friends!',
      subTitle: 'Shubh Festival Contest is here!',
      image: 'assets/icons/gift.png',
      actionText: 'Join Now',
      actionBgColor: const Color(0xff35BCAF),
      gradientColors: [
        const Color(0xff15C68B).withOpacity(0.9),
        const Color(0xff0288C3).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Lucky winners get ₹50',
      subTitle: 'Shubh Festival Contest is here!',
      image: 'assets/icons/bonus.png',
      actionText: 'Join Now',
      actionBgColor: const Color(0xff9771BB),
      gradientColors: [
        const Color(0xff8591F3).withOpacity(0.9),
        const Color(0xffFA75B0).withOpacity(0.7),
      ],
    ),
    DataModel(
      title: 'Vote for your fav Game!',
      subTitle: 'Whats the best Game in MGPL App?',
      image: 'assets/icons/analytics.png',
      actionText: 'Vote Now',
      actionBgColor: const Color(0xffB2615D),
      gradientColors: [
        const Color(0xffEC8A6D).withOpacity(0.9),
        const Color(0xffED709B).withOpacity(0.7),
      ],
    ),
  ];
}
